csharp/ql/integration-tests/all-platforms/msbuild/test.py

```diff
@@ -1,6 +1,24 @@
+import os
+
 from create_database_utils import *
 from diagnostics_test_utils import *
 
+def is_running_on_apple_silicon():
+   arch = subprocess.Popen(['sysctl', 'machdep.cpu.brand_string'], stdout=subprocess.PIPE)
+   output, errors = arch.communicate()
+   if b'apple' in output.lower():
+       return True
+   return False
+
+# if on ARM runners, remove Mono from the path, so we're using
+# dotnet restore instead of nuget.exe restore - on ARM machines
+# we run dotnet msbuild (instead of the mono-provided msbuild.exe)
+# so we need to match the restore command, too.
```

Comment:
Why don't we change the autobuilder instead, so it runs `dotnet restore` instead of `nuget.exe`?

Reply:
It's happening here: https://github.com/github/codeql/pull/14789